RAT/utils/plotting.py

```diff
@@ -154,25 +153,29 @@ def plot_ref_sld(data: PlotEventData, fig: Figure = None, delay: bool = True):
             plot_errorbars(ref_plot, sd_x_s, sd_y_s, sd_e_s, True, color)
 
         # Plot the slds on plot (1,2)
-        for j in range(1, sld.shape[1]):
-            sld_plot.plot(sld[:, 0],
-                          sld[:, j],
+        for j in range(len(sld)):
+            sld_plot.plot(sld[j][:, 0],
+                          sld[j][:, 1],
                           label=f'sld {i+1}',
-                          color=color,
-                          linewidth=2)
+                          linewidth=1)
 
         if data.resample[i] == 1 or data.modelType == 'custom xy':
-            new = makeSLDProfileXY(layer[0, 1],
-                                   layer[-1, 1],
-                                   data.subRoughs[i],
-                                   layer,
-                                   len(layer),
-                                   1.0)
-
-            sld_plot.plot([row[0]-49 for row in new],
-                          [row[1] for row in new],
-                          color=color,
-                          linewidth=1)
+            layers = data.resampledLayers[i][0]
+            for j in range(len(data.resampledLayers[i])):
+                layer = data.resampledLayers[i][j]
+                if layers.shape[1] == 4:
+                    layer = np.delete(layer, 2, 1)
+                new_profile = makeSLDProfileXY(layers[0, 1],      # Bulk In
+                                               layers[-1, 1],     # Bulk Out
```

Comment:
The code you sent me to address this had this line as: "layers[-1, 2]" - could you double check which it should be for me please.